crates/network/src/routing/mod.rs

```diff
@@ -147,12 +128,8 @@ impl<
 where
     ConsensusMsg: Debug + Send + Sync + 'static,
     ShuffleIn: Debug + Send + Sync + 'static,
-    ShuffleOut: TargetConsensusOrIngress<ShuffleToCon, ShuffleToIngress>,
-    ShuffleToCon: WithPartitionKey + Into<ConsensusMsg> + Debug,
-    ShuffleToIngress: Into<IngressIn> + Debug,
-    IngressOut: TargetConsensusOrShuffle<IngressToCon, IngressToShuffle>,
-    IngressToCon: WithPartitionKey + Into<ConsensusMsg> + Debug,
-    IngressToShuffle: TargetShuffle + Into<ShuffleIn> + Debug,
+    ShuffleOut: WithPartitionKey + Into<ConsensusMsg> + Debug,
```

Comment:
I wonder if this still needs to be generic. Can this be replaced by Envelope?

Reply:
I think you are right. I'll try to further simplify :-)